File: job11/pc.py
def produce():
    print("PRODUCE")

    print("produce 0")
    yield 0

    print("produce 1")
    yield 1

    print("produce 2")
    yield 2

    print("produce 3")
    yield 3

    print("produce 4")
    yield 4

    print("produce 5")
    yield 5

    print("produce 6")
    yield 6

    print("produce 7")
    yield 7

def consume():
    print("\tCONSUME")

    x = next(coro_p)
    print("\tconsume %d" % x)

    x = next(coro_p)
    print("\tconsume %d" % x)

    x = next(coro_p)
    print("\tconsume %d" % x)

    x = next(coro_p)
    print("\tconsume %d" % x)

    x = next(coro_p)
    print("\tconsume %d" % x)

    x = next(coro_p)
    print("\tconsume %d" % x)

    x = next(coro_p)
    print("\tconsume %d" % x)

    x = next(coro_p)
    print("\tconsume %d" % x)

    print("END")

coro_p = produce()
consume()

File: job11/pc.c
#include <stdio.h>
#include <stdlib.h>
#include "lib/coro.h"

coro_t *coro_p;
coro_t *coro_c;

void produce()
{
    puts("PRODUCE");

    printf("produce 0\n");
    coro_yield(0);

    printf("produce 1\n");
    coro_yield(1);

    printf("produce 2\n");
    coro_yield(2);

    printf("produce 3\n");
    coro_yield(3);

    printf("produce 4\n");
    coro_yield(4);

    printf("produce 5\n");
    coro_yield(5);

    printf("produce 6\n");
    coro_yield(6);

    printf("produce 7\n");
    coro_yield(7);

    puts("NEVER REACH HERE");
}

void consume()
{
    int x;

    puts("\tCONSUME");

    x = coro_resume(coro_p);
    printf("\tconsume %d\n", x);

    x = coro_resume(coro_p);
    printf("\tconsume %d\n", x);

    x = coro_resume(coro_p);
    printf("\tconsume %d\n", x);

    x = coro_resume(coro_p);
    printf("\tconsume %d\n", x);

    x = coro_resume(coro_p);
    printf("\tconsume %d\n", x);

    x = coro_resume(coro_p);
    printf("\tconsume %d\n", x);

    x = coro_resume(coro_p);
    printf("\tconsume %d\n", x);

    x = coro_resume(coro_p);
    printf("\tconsume %d\n", x);

    puts("END");
    exit(0);
}

int main()
{
    coro_p = coro_new(produce);
    coro_c = coro_new(consume);
    coro_boot(coro_c);
    return 0;
}
